datalad/interface/clean.py

```diff
@@ -35,30 +44,64 @@ class Clean(Interface):
     """
 
     _params_ = dict(
+        dataset=Parameter(
+            args=("-d", "--dataset"),
+            # TODO: this probably changes to install into the dataset (
+            # add_to_super)
+            # and to install the thing 'just there' without operating 'on' a
+            # dataset.
+            # Adapt doc.
+            # MIH: `shouldn't this be the job of `add`?
+            doc="""specify the dataset to perform the clean operation on.  If
+                no dataset is given, an attempt is made to identify the dataset
+                in current working directory""",
+            constraints=EnsureDataset() | EnsureNone()),
         # TODO:  --info  -- which performs dry run just summarizing what is to be cleaned up
         # TODO: Python only???
-        annex=Parameter(
-            doc="""annex instance to use"""
-            #constraints=EnsureStr() | EnsureNone()
-        ),
+        what=Parameter(
+            args=("--what",),
+            dest='what',
+            choices=('cached-archives', 'annex-tmp'),
+            nargs="*",
+            doc="""What to clean.  If none specified -- all known targets are
+            cleaned"""),
+        recursive=recursion_flag,
+        recursion_limit=recursion_limit,
     )
 
     @staticmethod
-    def __call__(annex=None):
-        if annex is None:
-            annex = get_repo_instance(class_=AnnexRepo)
-        topdir = opj(annex.path, ARCHIVES_TEMP_DIR)
-        temp_archives_dirs = glob(opj(annex.path, ARCHIVES_TEMP_DIR, '*'))
-        if temp_archives_dirs:
-            pl = len(temp_archives_dirs) > 1
-            pl1, pl2 = ('s', 'ies') if pl else ('', 'y')
+    @datasetmethod(name='clean')
+    def __call__(dataset=None, what=None, recursive=False, recursion_limit=None):
+        if dataset is None:
+            dataset = require_dataset(dataset, purpose='clean-up')
+
+        for dirpath, flag, msg, sing_pl in [
+            (ARCHIVES_TEMP_DIR, "cached-archives",
+             "temporary archive", ("directory", "directories")),
+            (ANNEX_TEMP_DIR, "annex-tmp",
+             "temporary annex", ("file", "files")),
+        ]:
+            lgr.info("Considering to clean %s:%s", dataset, dirpath)
+            if not ((what is None) or (flag in what)):
+                continue
+            topdir = opj(dataset.path, dirpath)
+            paths = glob(opj(topdir, '*'))
+            if paths:
+                pl = len(paths) > 1
+                pwd = getpwd()
```

Comment:
old code I forgot to RF since it provided sensible output ;)
I guess it is worth discussing what our UI should strive to output -- I often get overwhelmed with full paths polluting my terminal while working within some reasonably lengthy path and already knowing that I am in that bloody dataset.  So here it was providing output relative to curdir since that is what you expect to get most of the time (e.g. when you do `ls file` you don't get a full path to the file). and if was outside of that path -- would have provided a full path.  `datalad ls` also does it to some degree... but indeed we might want to strive to unify/optimize display of such output to the user